### PR TITLE
Replace ActionableMessage components with BannerAlerts in SIWE Sign-in with Ethereum page

### LINKED ISSUE #18206: [Bug] ActionableMessage component styles do not support Icon component (.mm-icon)
### Describe the bug

ActionableMessage component styles do not support Icon component

### Before
Screen Shot 2023-03-16 at 4 49 31 PM

### After
Screen Shot 2023-03-16 at 4 29 28 PM

### Steps to reproduce

1. Login to the extension
2. Navigate to the test dapp
3. Click Sign In With Ethereum (Bad Account)
4. Scroll to the bottom and observe the warning

Check other ActionableMessage components 

### Error messages or log output

_No response_

### Version

10.26.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Fixes #18206

- .mm-icon Icon component uses a <span> element which was not supported by ActionableMessage
- Replace SIWE ActionableMessage usage with BannerAlert
  - text in alerts are slightly larger due to the change
- Add `@deprecate` jsdoc comment to ActionableMessage 

### Before (v10.26.1)
(actually, the icon locations were temp fixed in this hotfix #18200 just before this PR.)
Screen Shot 2023-03-16 at 4 49 31 PM
Screen Shot 2023-03-17 at 11 07 27 AM

### After
Screen Shot 2023-03-17 at 10 31 51 AM
Screen Shot 2023-03-17 at 11 16 44 AM

## Manual Testing Steps

Login to the extension
Navigate to the test dapp
Click Sign In With Ethereum (Bad Account)
Scroll to the bottom and observe the warning
Check other ActionableMessage components

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
We could deprecate `ActionableMessage` for `BannerAlert` would it be possible to add a deprecation message above the component as well as this fix?

in `ui/components/ui/actionable-message/actionable-message.js`

```
/**
 * @deprecated `<ActionableMessage />` has been deprecated in favour of the `<BannerAlert />` component in ./ui/components/component-library/banner-alert/banner-alert.js
 *
 * See storybook documentation for Text here https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-banneralert--default-story#banneralert
 *
 * Help to replace `ActionableMessage` with `BannerAlert` by submitting a PR
 */
```
Screenshot 2023-03-16 at 5 24 28 PM

Screenshot 2023-03-16 at 5 24 58 PM

File: ui/components/app/signature-request-siwe/signature-request-siwe.js [hunk 1]
```diff
@@ -101,38 +103,25 @@ export default function SignatureRequestSIWE({
       ) : null}
       <Message data={formatMessageParams(parsedMessage, t)} />
       {!isMatchingAddress && (
-        <ActionableMessage
-          className="signature-request-siwe__actionable-message"
-          type="warning"
-          message={t('SIWEAddressInvalid', [
+        <BannerAlert severity={SEVERITIES.WARNING}>
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:106]: Add margins 

Screenshot 2023-03-18 at 10 17 03 AM


```suggestion
        <BannerAlert severity={SEVERITIES.WARNING}
          marginLeft={4}
          marginRight={4}
          marginBottom={4}
        >
```

File: ui/components/app/signature-request-siwe/signature-request-siwe.js [hunk 2]
```diff
@@ -101,38 +103,25 @@ export default function SignatureRequestSIWE({
...
+        <BannerAlert
+          severity={SEVERITIES.DANGER}
+          marginLeft={4}
+          marginRight={4}
+          marginBottom={4}
+        >
+          <Text variant={TextVariant.bodyMdBold}>
+            {t('SIWEDomainInvalidTitle')}
+          </Text>{' '}
+          <Text variant={TextVariant.bodyMd}>{t('SIWEDomainInvalidText')}</Text>
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:123]: Note to DS team to allow for `<strong/> tags cc @garrettbear so we could do something like
```
<Text>
  <strong>{t('SIWEDomainInvalidTitle')}</strong> {t('SIWEDomainInvalidText')}
</Text>

Comment (georgewrmarshall) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:123]: nit: default variant for the `Text` component is `bodyMd` so shouldn't need variant here
```suggestion
          <Text>{t('SIWEDomainInvalidText')}</Text>
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking great! One small suggestion

File: ui/components/app/signature-request-siwe/signature-request-siwe.js (hunk 1 above)
Comment (digiwand) [reply ui/components/app/signature-request-siwe/signature-request-siwe.js:106]: oops, forgot to push up this change. Thank you!

Comment (digiwand) [reply ui/components/app/signature-request-siwe/signature-request-siwe.js]: updated https://github.com/MetaMask/metamask-extension/pull/18207/commits/ee0ce10ebcf951bbc05b18197ca41dda4a534f37

File: ui/components/app/signature-request-siwe/signature-request-siwe.js (hunk 2 above)
Comment (digiwand) [reply ui/components/app/signature-request-siwe/signature-request-siwe.js]: updated [9b98162](https://github.com/MetaMask/metamask-extension/pull/18207/commits/9b98162d652667f299d24cca71f4ed8d0269d5c9)

**Review by georgewrmarshall — COMMENTED** [review]
LGTM! Just need to fix a linting issue that was probably introduced by my suggestions apologies!

**Comment by digiwand** [comment]
Thanks @georgewrmarshall! fixed https://github.com/MetaMask/metamask-extension/pull/18207/commits/de6a131ddaedaab00abc52199508abf721653e99

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀

**Review by jpuri — APPROVED** [review]

**Pull request merged**